x/auth/ante/ext_test.go

```diff
@@ -0,0 +1,36 @@
+package ante_test
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
+)
+
+func (suite *AnteTestSuite) TestRejectExtensionOptionsDecorator() {
+	suite.SetupTest(true) // setup
+	suite.txBuilder = suite.clientCtx.TxConfig.NewTxBuilder()
+
+	reod := ante.NewRejectExtensionOptionsDecorator()
+	antehandler := sdk.ChainAnteDecorators(reod)
+
+	// no extension options should not trigger an error
+	theTx := suite.txBuilder.GetTx()
+	_, err := antehandler(suite.ctx, theTx, false)
+	suite.Require().NoError(err)
+
+	extOptsTxBldr, ok := suite.txBuilder.(tx.ExtensionOptionsTxBuilder)
```

Comment:
Shouldn't we ensure this always gets executed?

Reply:
What do you mean? This isn't needed if we're using an amino stdtx TxConfig

Reply:
It seems weird to me to skip an entire section of a test based on if a type matches an interface. My comment was in light of this, meaning how can we ensure this is always executed and not need to type check.

Reply:
Well even if we add it to the standard `TxBuilder` interface, `SetExtensionOptions` will need to error for `StdTx` in which case we shouldn't expect the decorator to return an error... either way there's some conditional behavior.

Reply:
Mind if we merge this as is @alexanderbez ? Like I said changing `TxBuilder` won't remove the conditional it will just add error handling elsewhere.

Reply:
Sure 👍